cmd/snap-confine/mount-support.c

```diff
@@ -712,3 +712,87 @@ void sc_ensure_shared_snap_mount()
 			    NULL);
 	}
 }
+
+static void sc_make_slave_mount_ns()
+{
+	if (unshare(CLONE_NEWNS) < 0) {
+		die("can not unshare mount namespace");
+	}
+	// In our new mount namespace, recursively change all mounts
+	// to slave mode, so we see changes from the parent namespace
+	// but don't propagate our own changes.
```

Comment:
This will break `udisks2` snap that can perform real mounts on the real outer namespace because /media is arranged specifically to allow this.

Reply:
I think that the decision to make everything slave mounts it a bit too strong. We could be more precise and thus save the special property of the `/media` directory. Then we don't have to say that mount namespace event propagation only works in certain cases.